engine/src/main/java/com/datatorrent/stram/StramLocalCluster.java

```diff
@@ -308,6 +313,24 @@ public StramLocalCluster(LogicalPlan dag) throws IOException, ClassNotFoundExcep
     this.dnmgr = new StreamingContainerManager(dag);
     this.umbilical = new UmbilicalProtocolLocalImpl();
 
+    Method addURL;
+    URLClassLoader loader;
+    try {
+      loader = (URLClassLoader)ClassLoader.getSystemClassLoader();
+      addURL = URLClassLoader.class.getDeclaredMethod("addURL", new Class[] { URL.class });
+      addURL.setAccessible(true);
+    } catch (NoSuchMethodException e) {
+      throw new RuntimeException("No method named addURL found.", e);
+    }
+
+    for (String jarPath : dag.getJarResources().keySet()) {
+      try {
+        addToClasspath(jarPath, loader, addURL);
+      } catch (InvocationTargetException | IllegalAccessException e) {
```

Comment:
@vrozov I think all the exceptions can be propagated to parent. In order not to change the interface of StramLocalCluster constructor, I've captured exceptions and throwing a RuntimeException for it.

@gauravgopi123 As the exception is passed to the caller now, try/catch is no more there for what you've pointed out.

@tweise I have to add the give jar to classpath hence accessing the protected method via Reflection. Also this is in Local mode and not in actual stram.